docs/user-guide/coordinate-transformations.ipynb

```diff
@@ -0,0 +1,359 @@
+   "execution_count": null,
+   "id": "be50be41-9486-4943-99da-99653639ea98",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def x_square(x):\n",
+    "    return x * x"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f22ec1c0-ad99-418b-820a-a6dad6fb82aa",
+   "metadata": {},
+   "source": [
+    "Next, we create a `dict`, mapping from an output coord name to a function that can create this coordinate.\n",
+    "The [sc.show_graph](../generated/functions/scipp.show_graph.rst#scipp.show_graph) helper is a convient tool for visualizing the coordinate transformation defined by such as mapping:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "33f5e9ef-35a8-49e6-9731-8b33a525c8fb",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "graph = {'x^2':x_square}\n",
+    "sc.show_graph(graph)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a658c4ce-ab11-41ed-84c5-af8df71838ab",
+   "metadata": {},
+   "source": [
+    "Here, the `x` coordinate can be consumed by the `x_square` function, creating the `x^2` coordinate.\n",
+    "Note that the function name and coordinate are unrelated.\n",
+    "Next, we can call `transform_coords`.\n",
+    "Apart from the graph, we also pass a list of desired output coordinates, here simply `['x^2']`.\n",
+    "`transform_coords` returns a new (shallow-copied) data array with added coordinates:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ba462ced-c928-4f02-b60d-af8a602c03dd",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "transformed = da.transform_coords(['x^2'], graph=graph)\n",
+    "transformed"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ab752aee-f2d7-4cae-8c7a-b33a2f7e8033",
+   "metadata": {},
+   "source": [
+    "Note how `x` is now an attribute, i.e., operations will not use it for alignment anymore.\n",
+    "This is important since it will allow for operations combining `transformated` with other data that may have matching `x^2` but not `x`.\n",
+    "\n",
+    "### Example: Multi-step transform splitting and combining input coords"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "833948c5-4657-43a7-bf38-46c385bd7e26",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import scipp as sc\n",
+    "import numpy as np\n",
+    "hour_steps = sc.arange(dim='timestamp', dtype='int64', unit='s', start=0, stop=3*24*60*60, step=60*60)\n",
+    "start = sc.scalar(np.datetime64('2021-01-01T17:00:00'))\n",
+    "datetime = start + hour_steps\n",
+    "\n",
+    "nsite = 1000\n",
+    "ntime = len(datetime)\n",
+    "# Note that these points are *note* uniformely distributed on a shpere, this is NOT a good way to\n",
+    "# generate such points\n",
+    "site = sc.vectors(dims=['cartesian'], values=np.random.rand(nsite, 3)) - sc.vector(value=[.5,.5,.5])\n",
+    "site /= sc.norm(site)\n",
+    "site *= 6.371 * sc.Unit('km')\n",
```

Comment:
What is `6.371`?

Reply:
The earth radius.

Reply:
Yes, didn't come to mind when I saw the number. Where is the missing factor of 1000 coming from?

Reply:
Probably because I copied this from somewhere with the wrong locale (where the meaning of `,` and `.` is swapped...), fixed, thanks!